geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/Z3Index.scala

```diff
@@ -0,0 +1,50 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.accumulo.index
+
+import org.apache.accumulo.core.data.{Mutation, Range}
+import org.geotools.factory.Hints
+import org.locationtech.geomesa.accumulo.data._
+import org.locationtech.geomesa.accumulo.index.AccumuloIndexAdapter.ScanConfig
+import org.locationtech.geomesa.accumulo.iterators.Z3Iterator
+import org.locationtech.geomesa.index.index.{Z3Index, Z3ProcessingValues}
+import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.filter.Filter
+
+// current version - normal table sharing
+case object Z3Index extends AccumuloFeatureIndex with AccumuloIndexAdapter
+    with Z3Index[AccumuloDataStore, AccumuloFeature, Mutation, Range] {
+
+  import org.locationtech.geomesa.utils.geotools.RichSimpleFeatureType.RichSimpleFeatureType
+
+  val Z3IterPriority = 23
+
+  override val version: Int = 4
```

Comment:
Ok... I think it see what's going on...  The current version won't be in the legacy package and the old ones will move there.